exercises/concept/weather-forecast-server/.meta/exemplar.go

```diff
@@ -0,0 +1,37 @@
+package weatherforecastserver
```

Comment:
@junedev 
To be honest I am having a bit of trouble completing this part.
So let me go through your comment step by step so I can make sure I understand every bit clearly:
1. > The general setup in this package should be similar to what I described for the client. You have a custom server struct (one field of which is the standard server):

`type WeatherServer struct{httpServer *http.Server, ...}`

2.  > with a New constructor:

`func NewWeatherServer() WeatherServer`

3. > and then that struct should have a Handle method that takes a path and a Handler function as parameters:

`func (ws *WeatherServer) Handle(path string, handler http.Handler)`

4. > adds those to the multiplexer:

So for this part I have a little bit of problem. First of all, since we want to pass this as the Handler for the server, this process should be done before defining the server in the `NewWeatherServer` function. The solution that comes to my mind is to keep an array of handlers in the WeatherServer structure and use the Handle to add to this array, and use the array to create the mux during initializing of the server. The problem with this approach is that we cannot use `Handle` before creating a `WeatherServer`, unless we do not initializ the `httpServer` in `NewWeatherServer`. I have looked at the example but they use `*httptreemux.ContextMux` as their mux type so that is not applicable to this solution. Did I understand this currectly and does my solution make sense for it?

5. > So the tasks for the student would be to set up the struct, constructor and Handle function (probably it makes sense to make this one task)

6. > then the next task is adding a GET route via the handle method:

Which is basically to compose a handler to feed to `Handle` function;

7. > and then the last task is adding the POST route via the handle method.

Same as 6.

I have looked for building web frameworks in Go, these are the best results that I came along, but none implemented things in the same manner:
https://go.dev/doc/articles/wiki/
https://semaphoreci.com/community/tutorials/building-go-web-applications-and-microservices-using-gin
https://blog.logrocket.com/creating-a-web-server-with-golang/